p2p/store.go

```diff
@@ -0,0 +1,43 @@
+package p2p
```

Comment:
store implementations (especially ephemeral ones) should live in an `internal` package, in this case `internal/p2putil`. We'll probably make them publicly visible eventually, but for now I want to keep the exposed API minimal.